src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp

```diff
@@ -1631,30 +1631,56 @@ void C2_MacroAssembler::minmax_fp_v(VectorRegister dst, VectorRegister src1, Vec
   vfadd_vv(dst, src2, src2, Assembler::v0_t);
 }
 
+// Set dst to NaN if any NaN input.
+void C2_MacroAssembler::minmax_fp_masked_v(VectorRegister dst_src1, VectorRegister src2,
+                                           bool is_double, bool is_min, int vector_length,
+                                           VectorRegister tmp1, VectorRegister tmp2, VectorRegister vmask) {
+  assert_different_registers(dst_src1, src2);
+
+  vsetvli_helper(is_double ? T_DOUBLE : T_FLOAT, vector_length);
+
+  // Check vector elements of src1 and src2 for quiet and signaling NaN.
+  vfclass_v(tmp1, dst_src1);
+  vfclass_v(tmp2, src2);
```

Comment:
As discussed offline, a better way for finding NaN from the vector elements is with `vmfne` instruction, like: `vmfeq.vv v0, va, va`. vmfne writes 1 to the destination element when the corresponding element of `va` is NaN.

Reply:
Fixed.